### PR TITLE
dns: document usage profiles & Android automatic mode

### PR DESCRIPTION
**Pull request by Mikaela**
## Description

Resolves: #1239

#### Check List 

- [x] I have read and understand [the contributing guidelines](https://github.com/privacytoolsIO/privacytools.io/blob/master/.github/CONTRIBUTING.md).

* Netlify preview for the mainly edited page:  https://deploy-preview-1242--privacytools-io.netlify.com/providers/dns/#icanndns

### CODE REVIEW

**Comment by Mikaela** [comment]
@madosss What do you think of this suggestion?

**Comment by Mikaela** [comment]
Oh, should https://tools.ietf.org/html/draft-ietf-dprive-dtls-and-tls-profiles-01 be linked somewhere?

File: _includes/sections/dns.html [hunk 1]
```diff
@@ -408,7 +408,11 @@ <h1 id="icanndns" class="anchor"><a href="#icanndns"><i class="fas fa-link ancho
 <h4>Terms</h4>
 
 <ul>
-  <li>DNS-over-TLS (DoT) - A security protocol for encrypted DNS on a dedicated port 853. Some providers support port 443 which generally works everywhere while port 853 is often blocked by restrictive firewalls.</li>
+  <li>DNS-over-TLS (DoT) - A security protocol for encrypted DNS on a dedicated port 853. Some providers support port 443 which generally works everywhere while port 853 is often blocked by restrictive firewalls. DoT has two modes:</li>
+    <ul>
+        <li>Oppurtunistic mode: the client attempts to form a DNS-over-TLS connection to the server on port 853 without performing certificate validation. If it fails, it will use unencrypted DNS. <span class="badge badge-warning" data-toggle="tooltip" data-original-title="In other words automatic mode leaves your DNS traffic vulnerable to SSLsrip and MITM attacks"><i class="fas fa-exclamation-triangle"></i></span></li>
```
Comment (nitrohorse) [inline _includes/sections/dns.html]: I’m wondering if there is a better place for this. Do these modes also apply to DoH? I’m thinking they would, but primarily related to Firefox's TRR about:config modes, yeah? Would it make sense to link the about:config section here under DoH?

Comment (Mikaela) [reply _includes/sections/dns.html]: No, the modes are part of *[Authentication and (D)TLS Profile for DNS-over-TLS and DNS-over-DTLS](https://tools.ietf.org/html/draft-ietf-dprive-dtls-and-tls-profiles-01)* for now which is one of the two DoT RFCs everyone ([Google Developer documentation](https://developers.google.com/speed/public-dns/docs/dns-over-tls)) cites.

I think this is currently a big benefit for DoT over DoH (but as you know DoH leads in censorship resistance), but this is hopefully changing as per *[Centralized DNS over HTTPS (DoH) Implementation Issues and Risks](https://tools.ietf.org/html/draft-ietf-dprive-dtls-and-tls-profiles-01)*.

Firefox's `about:config` or even `network.trr.mode` is just something Mozilla/Firefox has thought of, even if they may have ~~stolen~~ took inspiration from DoT.

Comment (nitrohorse) [reply _includes/sections/dns.html]: Ah gotcha, thanks for the clarity!

Comment (nitrohorse) [reply _includes/sections/dns.html]: Should we hyperlink the tooltip to point to SSLrip? Maybe to give users more info?

**Comment by Mikaela** [comment]
> Should we hyperlink the tooltip to point to SSLrip? Maybe to give users more info?

Any suggestions where to hyperlink it to?

**Comment by nitrohorse** [comment]
We mean sslstrip yeah? We could link to Moxie’s site: https://moxie.org/software/sslstrip/

**Comment by Mikaela** [comment]
> We mean sslstrip yeah? We could link to Moxie’s (author) site: https://moxie.org/software/sslstrip/

I am not entirely sure about that. Is there any more generic term for SSL removing attack or are the available words SSLstrip (typo fixed) and MITM attack? I haven't tested the tool, which only talks about https, but I think the principle is same.

**Comment by nitrohorse** [comment]
> > We mean sslstrip yeah? We could link to Moxie’s (author) site: https://moxie.org/software/sslstrip/
> 
> I am not entirely sure about that. Is there any more generic term for SSL removing attack or are the available words SSLstrip (typo fixed) and MITM attack? I haven't tested the tool, which only talks about https, but I think the principle is same.

Ah sorry; I think the vulnerability is just "SSL Strip." We could also link to Moxie's [overview video](https://invidio.us/watch?v=MFol6IMbZ7Y).

**Comment by Mikaela** [comment]
> Ah sorry; I think the vulnerability is just "SSL Strip." We could also link to Moxie's overview video.

I will fix the name, but I am not entirely happy about the idea of linking to that video either as it's so long and it's nothing specific to DoT. Could we leave it as it is as curious people can type it to their favourite search engine and learn a lot anyway?

File: _includes/sections/dns.html [hunk 2]
```diff
@@ -444,6 +448,9 @@ <h3>Worth Mentioning and Additional Information</h3>
   <li><strong>Encrypted DNS clients for mobile:</strong>
     <ul>
       <li><em>Android 9</em> comes with a DoT client by <a href="https://support.google.com/android/answer/9089903">default</a>. <span class="badge badge-warning" data-toggle="tooltip" data-original-title="...but with some caveats"><a href="https://www.quad9.net/private-dns-quad9-android9/"><i class="fas fa-exclamation-triangle"></i></a></span></li>
+        <ul>
+          <li>We recommend selecting <em>Private DNS provider hostname</em> and entering the DoT address from documentation of your DoT provider to enable strict mode (see Terms above). <span class="badge badge-warning" data-toggle="tooltip" data-original-title="If you are on a network blocking access to port 853, Android will error about the network not having internet connectivity."><i class="fas fa-exclamation-triangle"></i></span></li>
```
Comment (Mikaela) [inline _includes/sections/dns.html]:
```suggestion
          <li>We recommend selecting <em>Private DNS provider hostname</em> and entering the DoT address from documentation of your DoT provider in order to enable strict mode (see Terms above). <span class="badge badge-warning" data-toggle="tooltip" data-original-title="If you are on a network blocking access to port 853, Android will error about the network not having internet connectivity."><i class="fas fa-exclamation-triangle"></i></span></li>
```

Would this be more proper English, or is it fine?

**Comment by nitrohorse** [comment]
> Could we leave it as it is as curious people can type it to their favourite search engine and learn a lot anyway?

You’re right, yeah, agree 👍🏼

File: _includes/sections/dns.html (hunk 2 above)
Comment (nitrohorse) [reply _includes/sections/dns.html]: Either is fine 👍🏼

**Pull request merged**